src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +30,43 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] addressValues=removePrefix(trimmedAddress, ADDRESS_PREFIX).split(",", 4);
+
+        if (addressValues.length>=1){
```

Comment:
you can think of splitting into parts, i am not sure you will find any library function that will do exactly what you want here!